store/structured/mongo/test/test.go

```diff
@@ -30,8 +34,32 @@ var _ = ginkgo.AfterSuite(func() {
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 	err = client.Database(database).Drop(context.Background())
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	runDeferredCleanups()
 })
 
+// DeferredCleanup provides a way to have something run during AfterSuite.
+//
+// Ginkgo only allows a single AfterSuite to be instantiated, and it's here in this file, so
+// give other tests a chance to make use of it.
+func DeferredCleanup(f func()) {
+	cleanupHandlersMu.Lock()
+	defer cleanupHandlersMu.Unlock()
+	if cleanupHandlers == nil {
+		cleanupHandlers = []func(){}
+	}
```

Comment:
I believe `append` will do this for you?

Reply:
> I believe `append` will do this for you?

True, but this code was removed later, so it doesn't matter this time.